src/plot_api/plot_schema.js

```diff
@@ -208,6 +216,182 @@ exports.findArrayAttributes = function(trace) {
     return arrayAttributes;
 };
 
+/*
+ * Find the valObject for one attribute in an existing trace
```

Comment:
Is a `valObject` just a value? Or is it a nested property object? Or something else?

Reply:
Less vaguely, `valObjects` are defined [here](https://github.com/plotly/plotly.js/blob/7ea0d250ab6731be0b98d40ee0fd9e6bf667986c/src/lib/coerce.js#L21), but I'm trying to process how the return value relates to those definitions.

Reply:
hah yeah, `coerce.valObjects` *describes* `valObject`s but it isn't itself at set of `valObject`s. What we mean by it in this context is a schema entry describing one attribute. I'll see what I can do in terms of documenting this better...

Reply:
Not a big deal. Just another part of the code I haven't gone too deep in. To clear up a common misconception though, naming in CS is a very easy problem (randomly generated strings will do). Naming _for maximum interpretability_ is the hard part. 😄

Reply:
I feel like `coerce.valObjects` should be named `coerce.valObjectMeta` instead. To me and to [`PlotSchema.isValObject`](https://github.com/plotly/plotly.js/blob/11bb997e78128b9f47db2ed392982f82a026e589/src/plot_api/plot_schema.js#L130) a val object is any object with a `valType` key - which is essentially what `getTraceValObject` and `getLayoutValObject` return.

Reply:
`valObjectMeta` - and descriptions for `editType` and `impliedEdits` that make their way into the schema - in https://github.com/plotly/plotly.js/pull/1999/commits/96cc57fe5a7ce905901ecf18250ddcc2cc0c66f8